services/venice-server/src/main/java/com/linkedin/venice/listener/ReadQuotaEnforcementHandler.java

```diff
@@ -312,6 +313,8 @@ private void updateQuota(PartitionAssignment partitionAssignment) {
           "Routing data changed on quota enforcement handler with 0 replicas assigned to this node, removing quota for resource: {}",
           topic);
       storeVersionRateLimiters.remove(topic);
+      stats.getStoreStats(Version.parseStoreFromKafkaTopicName(topic))
+          .removeVersion(Version.parseVersionFromKafkaTopicName(topic));
```

Comment:
Nit,
Can you parse store-name/version from the topic name once and reuse them in the whole function?